### DISCUSSION TITLE
Can create pool on designated osd with rook-ceph?

### DISCUSSION THREAD

**Discussion by zhucan** [rook/rook#9749] [Q&A]
@leseb

**Comment by leseb** [comment]
See `deviceClass` in https://rook.io/docs/rook/v1.8/ceph-pool-crd.html#pool-settings and `crushDeviceClass` in https://rook.io/docs/rook/v1.8/ceph-cluster-crd.html#other-status

**Comment by zhucan** [comment]
@leseb  When creating the pool，it can only use all of ssd osd or hdd osd, but can not use a few osd. Is it like this?

**Reply by leseb** [reply]
To isolate OSD you can use deviceFilter see https://rook.io/docs/rook/v1.8/ceph-cluster-crd.html#storage-selection-settings

**Reply by zhucan** [reply]
I means how to use "crush rules"  to create pools with designated osd.

**Reply by zhucan** [reply]
we can't create "crush rules" by ourself. @leseb

**Reply by travisn** [reply]
@zhucan You don't need to create custom crush rules. Basically, you need to:
1. Create the OSDs with the correct deviceClass settings
2. View that the deviceClass is correct for ceph (shown with a command such as `ceph osd tree` in the toolbox)
3. In the pool settings, specify the [deviceClass](https://github.com/rook/rook/blob/master/deploy/examples/pool.yaml#L33) where you want the data to be stored.

**Comment by zhucan** [comment]
@leseb so we need to create  the deviceClass manually. Maybe it's better to create it by crd?
